lib/moon/packages/debug/debug_shell.rb

```diff
@@ -58,38 +113,53 @@ def initialize_members(font)
     @history = []
     @history_index = 0
     @contents = []
-    @input_text = Moon::Text.new("", font)
-    @log_text = Moon::Text.new("", font)
+    @input_text = Moon::Label.new("", font)
+    @log_text = Moon::Label.new("", font)
     @log_text.line_h = 1
     @context = DebugContext.new
 
     @input_text.color = Moon::Vector4.new(1, 1, 1, 1)
     @log_text.color = Moon::Vector4.new(1, 1, 1, 1)
 
-    @input_background.w = w
-    @input_background.h = h
-    @seperator.w = w
-    @seperator.h = 1
+    on :resize do
+      @input_background.w = w
+      @input_background.h = h
+      @seperator.w = w
+      @seperator.h = 1
 
-    @log_text.position.set(4, 4-8, 0)
-    @input_text.position.set(4,5*h/6+4-8, 0)
-    @seperator.position.set(0, @input_text.y, 0)
-    @caret.position.set(@input_text.x, @input_text.position.y+2, 0)
+      @log_text.position.set(4, -4, 0)
+      @input_text.position.set(4, h - @input_text.font.size - 8, 0)
+      @seperator.position.set(0, @input_text.y, 0)
```

Comment:
*separator. http://www.dailywritingtips.com/forum/showthread.php?453-SEPARATE-and-SEPERATE